util/Migrator/DbMigrator.cs

```diff
@@ -34,15 +34,16 @@ public DbMigrator(string connectionString, ILogger<DbMigrator> logger)
 
             using (var connection = new SqlConnection(_masterConnectionString))
             {
+                var databaseName = new SqlConnectionStringBuilder(_connectionString).InitialCatalog;
                 var command = new SqlCommand(
-                    "IF ((SELECT COUNT(1) FROM sys.databases WHERE [name] = 'vault') = 0) " +
-                    "CREATE DATABASE [vault];", connection);
+                    "IF ((SELECT COUNT(1) FROM sys.databases WHERE [name] = '" + databaseName + "') = 0) " +
```

Comment:
Maybe moot in this context/setting, but it feels like we should parameterize this query vs. using string concatenation. Even if we're not concerned over SQL Injection risk coming from the configuration JSON, it would help mitigate other issues like someone with a database name of `Chad's_VaultDb` or something like that... 'cause why not AND most security source/code scanning tools are going to throw false negatives on this for SQL Injection.

Reply:
You cannot use parameters with the `CREATE DATABASE` command unless you do some Dynamic SQL magic.  We might be able to use parameters in the other two SQL statements. I'm not well versed enough to know if continuing with string concatenation is going to be an issue.

Reply:
Okay duh, sorry... but we then should be using escaped quoted identifiers with dynamic SQL in this case, see: https://stackoverflow.com/questions/45814069/escape-database-object-names and https://docs.microsoft.com/en-us/dotnet/api/system.data.common.dbcommandbuilder.quoteidentifier?view=netcore-3.1#System_Data_Common_DbCommandBuilder_QuoteIdentifier_System_String_

Reply:
here's a better option / explanation or perhaps easier means of safety checking: https://stackoverflow.com/a/9651743/14067465. Let me know if this is something I can help with.

Reply:
I parameterized the database name in the `WHERE` clauses and used a `SqlCommandBuilder` object to get the quoted identifier version of the database name (which ends up wrapping the name in square brackets).  I then use this quoted identifier in the `CREATE DATABASE` and `ALTER DATABASE` commands.  Does this cover the SQL injection attack vector?

Reply:
Yes, thank you, that should do it!